src/pe_source/data/pe_db/db_query_source.py

```diff
@@ -417,3 +416,82 @@ def org_root_domains(conn, org_uid):
     """
     df = pd.read_sql_query(sql, conn, params={"org_id": org_uid})
     return df
+
+
+def insert_intelx_breaches(df):
+    """Insert intelx breach data."""
+    conn = connect()
+    table = "credential_breaches"
+    # Create a list of tuples from the dataframe values
+    tuples = [tuple(x) for x in df.to_numpy()]
+    # Comma-separated dataframe columns
+    cols = ",".join(list(df.columns))
+    # SQL query to execute
+    query = """INSERT INTO {}({}) VALUES %s
+    ON CONFLICT (breach_name) DO UPDATE SET
+    password_included = EXCLUDED.password_included;"""
+    cursor = conn.cursor()
+    try:
+        extras.execute_values(
+            cursor,
+            query.format(
+                table,
+                cols,
+            ),
+            tuples,
+        )
+        conn.commit()
+        LOGGER.info("Successfully inserted/updated breaches into PE database.")
```

Comment:
Consider making this log message more specific:
```suggestion
        LOGGER.info("Successfully inserted/updated IntelX breaches into PE database.")
```